src/Search.elm

```diff
@@ -148,10 +149,25 @@ init args model =
                 |> fromSortId
                 |> Maybe.withDefault Relevance
       }
+        |> ensureLoading
     , Browser.Dom.focus "search-query-input" |> Task.attempt (\_ -> NoOp)
     )
 
 
+shouldLoad : Model a -> Bool
+shouldLoad model =
+    model.result == RemoteData.Loading
+
+
+ensureLoading : Model a -> Model a
+ensureLoading model =
+    if model.query /= Nothing && model.query /= Just "" && List.member model.channel channels then
+        { model | result = RemoteData.Loading }
+
+    else
+        model
```

Comment:
this is the only place where we explicitely define in which case we are able to load the data.

Reply:
I think some condition is missing since we shouldn't load when we open details of an search result item.

Reply:
in that case we simply don't call this function.